Errors in ODataConnectedService.Tests project

### Describe the bug
There are a few identical errors occured in `ODataConnectedService.Tests` project:

In `CodeGenCommandlineSetNamespacePrefixRepeatWithSchemaNameSpace`, `CodeGenCommandlineSetNamespacePrefixWithSingleNamespace` and `CodeGenCommandlineSimpleEdmx` tests:

> Message: Метод теста ODataConnectedService.Tests.ODataT4CodeGeneratorTests.CodeGenCommandlineSetNamespacePrefixRepeatWithSchemaNameSpace создал исключение: 
    FluentAssertions.Execution.AssertionFailedException: Expected empty string, but found "c:\<path>\AppData\Local\Temp\ae4fnt2w.0.cs(2921,29) : error CS1056: Компиляция преобразования: Непредусмотренный символ \"$\"\r\n-out \"d:\<path>\ODataConnectedService\test\ODataConnectedService.Tests\bin\Debug\test_out.output\" -a !!MetadataDocumentUri!File%3A%5Cd%3A%5CWork%5CMy1%5C1%5CODataConnectedService%5Ctest%5CODataConnectedService.Tests%5Cbin%5CDebug%5Ctest_edmx.edmx -a !!UseDataServiceCollection!False -a !!TargetLanguage!CSharp -a !!NamespacePrefix!NamespacePrefixRepeatWithSchemaNameSpace.Foo1.Foo11.Foo12 -p \"d:\<path>\ODataConnectedService\test\ODataConnectedService.Tests\bin\Debug\" \"d:\<path>\ODataConnectedService\test\ODataConnectedService.Tests\bin\Debug\ODataT4CodeGenerator.tt\""..

#### Version of the Project affected
ODataConnectedService v0.5.1
VisualStudio Version v16.4.3
Microsoft.Odata.Client v7.6.3

#### To Reproduce
1. Run `CodeGenCommandlineSetNamespacePrefixRepeatWithSchemaNameSpace`, `CodeGenCommandlineSetNamespacePrefixWithSingleNamespace` and `CodeGenCommandlineSimpleEdmx` tests
2. See errors

#### Expected behavior
Tests passed successfully.

#### Actual behavior
Tests occur the error.

#### Screenshots
![errors](https://user-images.githubusercontent.com/29679226/76619547-37648880-653c-11ea-9721-8674f9ef8e1e.png)

#### Solution
Probably, to solve this errors, should change in 2921 line of `ODataT4CodeGenerator.ttinclude` file:

```csharp
throw new Exception($"Could not get CLR type name for EDM type '{edmTypeReference.FullName()}'");
```

on

```csharp
throw new Exception(string.Format(CultureInfo.InvariantCulture, "Could not get CLR type name for EDM type '{0}'", edmTypeReference.FullName()));
```

--- a/test/ODataConnectedService.Tests/Templates/ODataT4CodeGeneratorTests.cs
+++ b/test/ODataConnectedService.Tests/Templates/ODataT4CodeGeneratorTests.cs
@@ -52,13 +52,39 @@ namespace ODataConnectedService.Tests
             MetadataUri = Uri.EscapeDataString(MetadataUri);
 
             string commonProgramFiles = Environment.GetEnvironmentVariable("CommonProgramFiles");
+            var programFiles = Environment.GetEnvironmentVariable("ProgramFiles");
             string T4TransformToolPathVer11 = commonProgramFiles + "\\Microsoft Shared\\TextTemplating\\14.0\\TextTransform.exe";
             string T4TransformToolPathVer12 = commonProgramFiles + "\\Microsoft Shared\\TextTemplating\\15.0\\TextTransform.exe";
-            string T4TransformToolPathVSVer2019 = "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\Common7\\IDE\\TextTransform.exe";
-            
-            if (File.Exists(T4TransformToolPathVSVer2019))
+            string T4TransformToolPathVSVer2019Enterprise = programFiles + "\\Microsoft Visual Studio\\2019\\Enterprise\\Common7\\IDE\\TextTransform.exe";
+            var T4TransformToolPathVSVer2019Professional = programFiles + "\\Microsoft Visual Studio\\2019\\Professional\\Common7\\IDE\\TextTransform.exe";
+            var T4TransformToolPathVSVer2019Community = programFiles + "\\Microsoft Visual Studio\\2019\\Community\\Common7\\IDE\\TextTransform.exe";
+            var T4TransformToolPathVSVer2017Enterprise = programFiles + "\\Microsoft Visual Studio\\2017\\Enterprise\\Common7\\IDE\\TextTransform.exe";
+            var T4TransformToolPathVSVer2017Professional = programFiles + "\\Microsoft Visual Studio\\2017\\Professional\\Common7\\IDE\\TextTransform.exe";
+            var T4TransformToolPathVSVer2017Community = programFiles + "\\Microsoft Visual Studio\\2017\\Community\\Common7\\IDE\\TextTransform.exe";
+
+            if (File.Exists(T4TransformToolPathVSVer2019Enterprise))
             {
-                T4TransformToolPath = T4TransformToolPathVSVer2019;
+                T4TransformToolPath = T4TransformToolPathVSVer2019Enterprise;
+            }
+            else if (File.Exists(T4TransformToolPathVSVer2019Professional))
+            {
+                T4TransformToolPath = T4TransformToolPathVSVer2019Professional;
+            }
+            else if (File.Exists(T4TransformToolPathVSVer2019Community))
+            {
+                T4TransformToolPath = T4TransformToolPathVSVer2019Community;
+            }
+            else if (File.Exists(T4TransformToolPathVSVer2017Enterprise))
+            {
+                T4TransformToolPath = T4TransformToolPathVSVer2017Enterprise;
+            }
+            else if (File.Exists(T4TransformToolPathVSVer2017Professional))
+            {
+                T4TransformToolPath = T4TransformToolPathVSVer2017Professional;
+            }
+            else if (File.Exists(T4TransformToolPathVSVer2017Community))
+            {
+                T4TransformToolPath = T4TransformToolPathVSVer2017Community;
             }
             else if(File.Exists(T4TransformToolPathVer11))
             {
@@ -140,7 +166,7 @@ namespace ODataConnectedService.Tests
         {
             string code = CodeGenWithT4Template(ODataT4CodeGeneratorTestDescriptors.SimpleMultipleFiles.Metadata, null, true, false, generateMultipleFiles : true);
             ODataT4CodeGeneratorTestDescriptors.SimpleMultipleFiles.Verify(code, true/*isCSharp*/, false/*useDSC*/);
-            
+
             string expectedTestType = ODataT4CodeGeneratorTest.NormalizeGeneratedCode(ODataT4CodeGeneratorTestDescriptors.GetFilecontent("SimpleMultipleTestType.cs"));
             string actualTestType = ODataT4CodeGeneratorTest.NormalizeGeneratedCode(File.ReadAllText(Path.Combine(Path.GetTempPath(), "TestType.cs")));
 
@@ -149,7 +175,7 @@ namespace ODataConnectedService.Tests
 
             string expectedCity = ODataT4CodeGeneratorTest.NormalizeGeneratedCode(ODataT4CodeGeneratorTestDescriptors.GetFilecontent("SimpleMultipleFilesCity.cs"));
             string actualCity = ODataT4CodeGeneratorTest.NormalizeGeneratedCode(File.ReadAllText(Path.Combine(Path.GetTempPath(), "City.cs")));
-            
+
             Assert.AreEqual(expectedTestType, actualTestType);
             Assert.AreEqual(expectedPersonGender, actualPersonGender);
             Assert.AreEqual(expectedCity, actualCity);
@@ -173,7 +199,7 @@ namespace ODataConnectedService.Tests
             var invalidEdmxDsvGreaterThanMdsv = @"<?xml version=""1.0"" standalone=""yes"" ?>
 <edmx:Edmx Version=""4.0"" xmlns:edmx=""http://docs.oasis-open.org/odata/ns/edmx"">
   <edmx:DataServices>
-    <Schema Namespace=""Test"" xmlns:d=""http://docs.oasis-open.org/odata/ns/data"" 
+    <Schema Namespace=""Test"" xmlns:d=""http://docs.oasis-open.org/odata/ns/data""
             xmlns=""http://docs.oasis-open.org/odata/ns/edm"">
       <EntityType Name=""TestType"">
         <Key>
@@ -195,7 +221,7 @@ namespace ODataConnectedService.Tests
                 Assert.AreEqual(ex.Message, "The value of the 'MaxDataServiceVersion' attribute must always be greater than or equal to the value of the 'OData-Version' attribute in the metadata document.");
             }
         }
-        
+
         [TestMethod]
         public void CodeGenSetNamespacePrefixWithSingleNamespace()
         {

